Firestore/src/Query.php

```diff
@@ -331,6 +346,10 @@ public function where($fieldPath, $operator, $value)
                 ]
             ];
         } else {
+            if (is_string($value) && $escapedPathString === self::DOCUMENT_ID) {
+                $value = $this->createDocumentReference($basePath, $value);
```

Comment:
`createDocumentReference` runs the `isDocument` check a second time. Since it is already done above on line 323, maybe we should extract the `isDocument` call out of `createDocumentReference`? That unfortunately means explicitly needing to call the `isDocument` check again prior to line 662.